tests/phpunit/CiviTest/CiviUnitTestCase.php

```diff
@@ -636,17 +638,18 @@ public function membershipStatusCreate($name = 'test member status') {
 
     $result = $this->callAPISuccess('MembershipStatus', 'Create', $params);
     CRM_Member_PseudoConstant::flush('membershipStatus');
-    return $result['id'];
+    return (int) $result['id'];
   }
 
   /**
+   * Delete the given membership status, deleting any memberships of the status first.
    * @param int $membershipStatusID
+   *
+   * @throws \CRM_Core_Exception
    */
-  public function membershipStatusDelete($membershipStatusID) {
-    if (!$membershipStatusID) {
-      return;
-    }
-    $result = $this->callAPISuccess('MembershipStatus', 'Delete', ['id' => $membershipStatusID]);
+  public function membershipStatusDelete(int $membershipStatusID) {
+    $this->callAPISuccess('Membership', 'get', ['status_id' => $membershipStatusID, 'api.Membership.get' => 1]);
+    $this->callAPISuccess('MembershipStatus', 'Delete', ['id' => $membershipStatusID]);
```

Comment:
Most of this PR looks cleaner/NFC -- eg the second call (`MembershipStatus.Delete`) looks essentially the same (omitting the unusued result variable).

I have to admit, it feels like I'm missing something - why add the call to `Membership.get` (chained with another `Membership.get`) if there's no output and no obvious side-effect? Is there some hidden side-effect of `Membership.get`?

Reply:
Nope - that looks like a typo - fixed